### ISSUE TITLE
Fossology 3.0.0 installation on VM, performance issues

### ISSUE THREAD

**Issue by kaiserxol** [fossology/fossology#683]
Hello,
I've used FOSSOlogy without problems, v. 2.0.0 for years in phisical server with 8 CPU core and 32 GB of ram, now we have installed the 3.0.0 version in a virtual machine with 4 core (Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz) and 32 GB of RAM. Now we have some problems performance when we upload new files in the new version, the tool became very very slow and if I try to enter in a project that have ended the upload process, the page will be very very slowly to respond.

I've some questions, if someone can help me:
- Is it correct to use 3.0.0 in a production environment? 
- 3.0.0 is a stable version?
- I can see in "top" command, always the postmaster with 99% of CPU using
![cattura](https://cloud.githubusercontent.com/assets/19837683/15929802/7de2182c-2e4f-11e6-8c77-38511546317e.PNG)
what do you think about? It's the less cpu core that can generate this problem or maybe we have installed a version not for production use?

If you need more details, just let me know.

Thanks to all in advance and best regards,
Kaiserxol

**Comment by mcjaeger** [comment]
How did you install it? was it from source or the package available at fossology.org/releases. I think the latter ones are broken.

Second, have you considered this page:
http://archive15.fossology.org/projects/fossology/wiki/SysConfig

Our fossology installation cannot know what system you have, so tuning is usually very necessary.

analogously I can write that with a similar E5 config, the tool is super fast ;-)

**Comment by mcjaeger** [comment]
and for your questions:
Is it correct to use 3.0.0 in a production environment? yes, but in your case I would prefer 3.1rc2
3.0.0 is a stable version? yes
I can see in "top" command, always the postmaster with 99% of CPU using. 
-> I have seen this too (with boost-1.5.8 for example) I have increased work mem to 32MB in postgresql setting.

**Comment by pguttmann** [comment]
You may also find some assistance from the scripts at:
https://github.com/fossology/fossology/tree/master/install/scripts
which assist with implementing the tuning referred to in the page Michael mentioned above ;)

**Comment by mcjaeger** [comment]
on a word on the 32MB work mem. better also to check the other postgresql setting, assume you have some idea what they mean. you could try the 32MB and then check if that runs better for you. then you might want to see if 8BM work mem is also working well, etc.

**Comment by mcjaeger** [comment]
maybe looking before writing: 
among the scripts Paul mentioned, the # ./pgsql-conf-fix.sh also modified the workmem for you.

**Comment by mcjaeger** [comment]
@kaiserxol, could we close the ticket / question answered? Kind regards, Michael

**Comment by mcjaeger** [comment]
I think feedback would be good, if there mentioned coverage of the FOSSology is solving performance concerns or more must be done. Keeping the question open for now, but I think after one week it makes sense to close this.

**Comment by kaiserxol** [comment]
Sorry for delay! Yes it work much better giving 32MB to postgres. I've also changed our import method, so now we doesn't import all project with all components with a single import (very very big 20/30 GB of data), but we import each component inside the project individually. The tool is very fast now!

Thanks also for the 2 links, "SYS" wiki and scripts, we have also made some tuning based on these guideline.

Best regards to both ;-)
